### ISSUE TITLE
Array value "null" is error in pre-processor

### ISSUE THREAD

**Issue by dcmartin** [emqx/kuiper#185]
If an specified `array` value in a stream is `null` the pre-processor indicates an invalid value; I have not checked to determine if a value of `[]` suffices to eliminate this error, but IMHO `null` should be a valid value for both `array` and `struct`; in my instance I have an `array` of `struct` of `string` and `bigint`

For example:
```
[{"count":5,"detected":[{"count":3,"entity":"car"},{"count":2,"entity":"truck"}],"event":{"camera":"woodshed","device":"ftpcams"}}]
[{"error":"error in preprocessor: invalid data type for detected, expect array but found <nil>(<nil>)"}]
[{"count":7,"detected":[{"count":1,"entity":"cow"},{"count":3,"entity":"dog"},{"count":2,"entity":"person"},{"count":1,"entity":"sheep"}],"event":{"camera":"sheshed","device":"netcams"}}]
[{"count":1,"detected":[{"count":1,"entity":"person"}],"event":{"camera":"foyer","device":"ftpcams"}}]

**Comment by jinfahua** [comment]
@dcmartin We'll fix the issue in the next release. Thanks for your feedback!

**Comment by ngjaying** [comment]
Released at v0.3.0
